### PR TITLE
Touch error filtering

### PR DESCRIPTION
**Pull request by yaffol**
This PR adds a touched ref that is bound to controls from
`useVuetifyControl` and a `handleBlur` composable that updates the
`touched` state after an input is blurred.

It also takes account of a new config setting `enableFilterErrorsBeforeTouch`
which allows the forcing of all errors to display, such as when a user
attempts to submit a form which is not completely valid.

### CODE REVIEW

File: vue-vuetify/src/complex/OneOfRenderer.vue [hunk 1]
```diff
@@ -123,14 +123,15 @@ const controlRenderer = defineComponent({
     const newSelectedIndex = ref(0);
     const dialog = ref(false);
     const t = useTranslator();
-
+    const { handleBlur } = useBlurHandler(control);
```
Comment (sdirix) [inline vue-vuetify/src/complex/OneOfRenderer.vue:126]: The `useBlurHandler` could also be made part of `useVuetifyControl`, then we wouldn't need to call it in every `setup` method.

File: vue-vuetify/src/util/composition.ts [hunk 2]
```diff
@@ -275,3 +296,13 @@ export const useNested = (element: false | 'array' | 'object'): NestedInfo => {
   }
   return nestedInfo;
 };
+
+
+export const useBlurHandler = (control: any) => {
```
Comment (sdirix) [inline vue-vuetify/src/util/composition.ts:301]: We should type the `control` here to at least declare the properties, see for example `useVuetifyLabel`

File: vue-vuetify/src/util/composition.ts [hunk 3]
```diff
@@ -87,6 +88,10 @@ export const useVuetifyControl = <
   const appliedOptions = useControlAppliedOptions(input);
   const isFocused = ref(false);
 
+  const filteredErrors = computed(() => {
+    return (touched.value || !input.control.value.config.enableFilterErrorsBeforeTouch)? input.control.value.errors : '';
```
Comment (sdirix) [inline vue-vuetify/src/util/composition.ts:92]: We could use the `appliedOptions` instead of the `config` so that the touched behavior can be enabled/disabled for specific fields via the ui schema.

**Review by sdirix — COMMENTED** [review]
I had a shallow look. Looks good to me for a first implementation.

One drawback which I noticed is that we don't have any clue about which errors are shown in child controls, so container renderers like object and arrays at the moment can't determine which errors to show in an error summary. Is that a use case for you?

File: vue-vuetify/src/complex/OneOfRenderer.vue (hunk 1 above)
Comment (yaffol) [reply vue-vuetify/src/complex/OneOfRenderer.vue]: Good idea! Moved `handleBlur` into `useVuetifyControl`.

File: vue-vuetify/src/util/composition.ts (hunk 2 above)
Comment (yaffol) [reply vue-vuetify/src/util/composition.ts]: Now that the renderers can access `handleBlur` directly from the `useVuetifyControl` there is no need to pass the control into the blur handler.

**Comment by yaffol** [comment]
> I had a shallow look. Looks good to me for a first implementation.
> 
> One drawback which I noticed is that we don't have any clue about which errors are shown in child controls, so container renderers like object and arrays at the moment can't determine which errors to show in an error summary. Is that a use case for you?

Thanks! Container errors are a use case, but as discussed handling the display of those errors seems quite complicated. 

Therefore I've updated the behaviour so that, when `enableFilterErrorsBeforeTouch` is true, container errors are not shown. When `enableFilterErrorsBeforeTouch` becomes false, all errors - including container errors - are shown.

This covers my primary use case of someone attempting to submit a form which is not valid, when they attempt to submit, all the errors are shown to them, so they can properly correct them.

**Comment by manuelmeister** [comment]
What is missing to get this merged? Can I help?

**Comment by sdirix** [comment]
Hi @manuelmeister, I'll review it very soon. My plan is to review, merge and release within the month. However as this is outside of [sponsored development](https://jsonforms.io/support), I can't make any promises for a delivery date.

In case you want to actively help: It's always useful to have additional testers who integrated the changes themselves and can confirm that they work. This gives considerably more confidence than when it's just me testing and making sure myself that I checked all edge cases.
